source/Octopus.Client/OctopusAsyncClient.cs

```diff
@@ -162,10 +164,21 @@ private bool IgnoreServerCertificateCallback(HttpRequestMessage message, X509Cer
         /// <returns>A fresh copy of the root document.</returns>
         public async Task<RootResource> RefreshRootDocument()
         {
-            RootDocument = await Get<RootResource>("~/api").ConfigureAwait(false);
+            RootDocument = await Get<RootResource>(rootDocumentUri).ConfigureAwait(false);
             return RootDocument;
         }
 
+        public async Task<IOctopusAsyncClient> ForSpaceContext(string spaceId)
+        {
+            this.clientOptions.SpaceId = spaceId;
```

Comment:
Don't mutate the existing `clientOption`.  This could have unintended side effects.  For example, if we needed to re-calculate the rootDocumentUri of the current client.